web/src/app/schedules/ScheduleRuleEditDialog.js

```diff
@@ -32,77 +32,47 @@ const mutation = gql`
   }
 `
 
-export default class ScheduleRuleEditDialog extends React.Component {
-  static propTypes = {
-    scheduleID: p.string.isRequired,
-    target: p.shape({
-      type: p.oneOf(['rotation', 'user']).isRequired,
-      id: p.string.isRequired,
-    }).isRequired,
-    onClose: p.func,
-  }
-
-  state = {
-    value: null,
-  }
-
-  shouldComponentUpdate(nextProps, nextState) {
-    if (this.state !== nextState) return true
-
-    return false
-  }
-
-  render() {
-    return (
-      <Query
-        query={query}
-        variables={{ id: this.props.scheduleID, tgt: this.props.target }}
-        noPoll
-        fetchPolicy='network-only'
-        render={({ data }) =>
-          this.renderMutation(data.schedule.target, data.schedule.timeZone)
-        }
-      />
-    )
-  }
+export default function ScheduleRuleEditDialog(props) {
+  const [value, setValue] = useState(null)
 
-  renderMutation(data, zone) {
-    return (
-      <Mutation mutation={mutation} onCompleted={this.props.onClose}>
-        {(commit, status) => this.renderDialog(data, commit, status, zone)}
-      </Mutation>
-    )
-  }
-
-  renderDialog(data, commit, status, zone) {
+  function renderDialog(data, commit, status, zone) {
     const defaults = {
-      targetID: this.props.target.id,
+      targetID: props.target.id,
       rules: data.rules.map((r) => ({
         id: r.id,
         weekdayFilter: r.weekdayFilter,
         start: gqlClockTimeToISO(r.start, zone),
         end: gqlClockTimeToISO(r.end, zone),
       })),
     }
+
+    function renderMutation(data, zone) {
+      return (
+        <Mutation mutation={mutation} onCompleted={props.onClose}>
+          {(commit, status) => renderDialog(data, commit, status, zone)}
+        </Mutation>
+      )
+    }
```

Comment:
The scope of this function will cause the build to fail- it should be placed underneath this function rather than inside of it